### ISSUE TITLE
ES6 class property lost when in module

### ISSUE THREAD

**Issue by pabigot** [jsdoc3/jsdoc#1182]
In the example below the instance property `p` is documented in the ES5 "class" but is missing from the documentation for the ES6 class.  If the `@module` tag is removed from the header it appears in both variants.

This may be related to the bug noted in #1134.

```
/**
 * @module
 */
'use strict';

/**
 * Test ES5 constructor handling.
 *
 * @param {String} p initializes {@link ES5#p}.
 *
 * @constructor
 */
function ES5(p) {
  /**
   * Some property.
   * @type {String}
   */
  this.p = p;
}

/**
 * Test ES6 constructor handling.
 */
class ES6 {
  /**
   * Create an ES6
   * @param {String} p initializes {@link ES6#p}.
   */
  constructor(p) {
    /**
     * Some property.
     * @type {String}
     */
    this.p = p;
  }
}
```

**Comment by pabigot** [comment]
Looking into it I see that doclets are being generated for both styles but with different longnames and properties.

With the ES5 syntax it's
* `module:mymod~ES5` kind=class scope=inner
* `module:mymod~ES5#p` kind=member scope=instance

With the ES6 syntax it's:
* `module:mymod~ES6` kind=class scope=inner
* `module:mymod~ES6.ES6` kind=class scope=static
* `module:mymod~ES6.ES6#p` kind=member scope=instance

If this namepath difference is not intentional I suspect the problem is in findSymbolMemberOf() directly or in Parser.resolveThis() when invoked from findSymbolMemberof().

Any suggestions on the best path to a fix?

**Comment by svenkreiss** [comment]
I see the same problem.
@pabigot Do you have a fix?

**Comment by bennyn** [comment]
Any update on this one? Because it blocks: https://github.com/englercj/tsd-jsdoc/issues/21

[tsd-jsdoc](https://github.com/englercj/tsd-jsdoc) is a tool to generate TypeScript definitions from JSDoc comments (pretty cool!), but when ES6 class properties are lost, then they are also lost in the definition file. 😢

**Comment by hegemonic** [comment]
Fixed on master. The fix will be included in JSDoc 3.5.0.
